code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -327,4 +327,77 @@ private static void failWithError(final AdobeCallback<?> callback, final AdobeEr
             callbackWithError.fail(error);
         }
     }
+
+    public static void displayPropositions(List<OptimizeProposition> propositions) {
+        trackWithData(
+                generateInteractionXdmForPropositions(
+                        propositions,
+                        OptimizeConstants.JsonValues.EE_EVENT_TYPE_PROPOSITION_DISPLAY));
+    }
+
+    private static Map<String, Object> generateInteractionXdmForPropositions(
```

Comment:
Considering this code is called from `Offer.java` as well, could you please move it to `OptimizeUtils`? Please also modify `Offer.java` to use the same utility method

Reply:
Please rename this to `generateDisplayXDM`

Reply:
done @spoorthipujariadobe